front/lib/api/files/preprocessing.ts

```diff
@@ -168,6 +170,59 @@ const extractTextFromPDF: PreprocessingFunction = async (
   }
 };
 
+// CSV preprocessing.
+// We upload the content of the CSV on the processed bucket and the schema in the snippet bucket.
+const extractContentAndSchemaFromCSV: PreprocessingFunction = async (
+  auth: Authenticator,
+  file: FileResource
+): Promise<Result<undefined, Error>> => {
+  try {
+    const readStream = file.getReadStream(auth, "original");
+    const processedWriteStream = file.getWriteStream(auth, "processed");
+    const schemaWriteStream = file.getWriteStream(auth, "snippet");
+
+    // Process the first stream for processed file
+    const processedPipeline = pipeline(
+      readStream.pipe(new PassThrough()),
+      processedWriteStream
+    );
+
+    // Process the second stream for snippet file
+    const snippetPipeline = pipeline(
+      readStream.pipe(new PassThrough()),
+      parse({
+        columns: true,
+        skip_empty_lines: true,
+        trim: true,
+      }),
+      analyzeCSVColumns,
+      async function* (schema) {
+        for await (const schemaData of schema) {
+          yield JSON.stringify(schemaData, null, 2);
```

Comment:
Actually we should not use async iterator here, as it could yields many chunks that we will try to stringified independently. Instead we should use the [Transform](https://nodejs.org/api/stream.html#streampipelinesource-transforms-destination-callback) stream to first collect, then stringify. This will remove the async iterator need on `analyzeCSVColumns`.

Example:
```
new Transform({
    decodeStrings: false, // Accept string input rather than Buffers
    construct(callback) {
      this.data = '';
      callback();
    },
    transform(chunk, encoding, callback) {
      this.data += chunk;
      callback();
    },
    flush(callback) {
      try {
        // Make sure is valid json.
        JSON.parse(this.data);
        this.push(this.data);
        callback();
      } catch (err) {
        callback(err);
      }
    },
  }),
```